cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -379,13 +401,41 @@ func (g *gitHubRepository) getReleaseByTag(ctx context.Context, tag string) (*gi
 	return release, nil
 }
 
-// downloadFilesFromRelease download a file from release.
-func (g *gitHubRepository) downloadFilesFromRelease(ctx context.Context, release *github.RepositoryRelease, fileName string) ([]byte, error) {
-	cacheID := fmt.Sprintf("%s/%s:%s:%s", g.owner, g.repository, *release.TagName, fileName)
-	if content, ok := cacheFiles[cacheID]; ok {
-		return content, nil
+// httpGetFilesFromRelease gets a file from github using http get.
+func (g *gitHubRepository) httpGetFilesFromRelease(ctx context.Context, version, fileName string) ([]byte, error) {
+	downloadURL := fmt.Sprintf("https://github.com/%s/%s/releases/download/%s/%s", g.owner, g.repository, version, fileName)
+	var retryError error
+	var content []byte
+	_ = wait.PollUntilContextTimeout(ctx, retryableOperationInterval, retryableOperationTimeout, true, func(ctx context.Context) (bool, error) {
+		resp, err := http.Get(downloadURL) //nolint:gosec,noctx
+		if err != nil {
+			retryError = errors.Wrap(err, "error sending request")
+			return false, nil
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			retryError = errors.Errorf("error getting file, status code: %d", resp.StatusCode)
+			return false, nil
+		}
+
+		body, err := io.ReadAll(resp.Body)
```

Comment:
```suggestion
		content, err = io.ReadAll(resp.Body)
```
Maybe, then you can drop l.428 (this aligns this func to what we do in downloadFilesFromRelease